File: src/main/java/com/fendyk/DTOs/MinecraftUserDTO.java
package com.fendyk.DTOs;

import java.util.ArrayList;
import java.util.Date;

public class MinecraftUserDTO {

    String id;
    String userId;
    Date authorizeExpireDate;
    Float quanta;
    ArrayList<TaggedLocationDTO> homes;
    ArrayList<String> memberLandIDs;
    ArrayList<SubscriptionRewardDTO> subscriptionRewards;
    LocationDTO lastLocation;

    public MinecraftUserDTO() {
        this.homes = new ArrayList<>();
        this.memberLandIDs = new ArrayList<>();
        this.subscriptionRewards = new ArrayList<>();
    }

    public ArrayList<SubscriptionRewardDTO> getSubscriptionRewards() {return subscriptionRewards;}


    public String getId() {
        return id;
    }


    public String getUserId() {
        return userId;
    }


    public Date getAuthorizeExpireDate() {
        return authorizeExpireDate;
    }


    public Float getQuanta() {
        return quanta;
    }


    public ArrayList<TaggedLocationDTO> getHomes() {
        return homes;
    }


    public ArrayList<String> getMemberLandIDs() {
        return memberLandIDs;
    }

    public LocationDTO getLastLocation() {
        return lastLocation;
    }
}

File: src/main/java/com/fendyk/DTOs/MinecraftUserDTO.kt
package com.fendyk.DTOs

data class MinecraftUserDTO(val id: String) {
    val userId: String? = null
    var authorizeExpireDate: String? = null
    var quanta: Double = 0.0
    var lastLocation: LocationDTO? = null
    var homes: ArrayList<TaggedLocationDTO> = ArrayList()
    var memberLandIDs: ArrayList<String> = ArrayList()
    var subscriptionRewards: ArrayList<SubscriptionRewardDTO> = ArrayList()
}
